MCprep_addon/materials/skin.py

```diff
@@ -703,6 +733,77 @@ def execute(self, context):
 		return {'FINISHED'}
 
 
+class MCPREP_OT_swap_skin_variant(bpy.types.Operator, spawn_util.VariationProp):
+	"""Apply the active UIlist skin to select characters"""
+	bl_idname = "mcprep.swap_skin"
+	bl_label = "Swap mob skin"
+	bl_description = "Swap the mobs variant"
+	bl_options = {'REGISTER', 'UNDO'}
+	
+	# def invoke(self, context, event):
+	# 	return context.window_manager.invoke_props_dialog(
+	# 		self, width=300 * util.ui_scale())
+
+	# def draw(self, context: Context):
+	# 	layout = self.layout
+	# 	self.draw_variation_ui(context, layout)
+	
+	@tracking.report_error
+	def execute(self, context: Context):
+		obj = context.object
+		if obj.type != 'ARMATURE':
+			self.report({'ERROR'}, "Please select an armature")
+			return {'CANCELLED'}
+		mats, skipped = getMatsFromSelected(obj, False)
+		mob_type = obj.get("MCPREP_mob_type", "Custom")
+		
+		if mob_type == "Custom":
+			self.report({'ERROR'}, "You are not allowed to use on Custom rig")
+			return {'CANCELLED'}
+		
+		texture_paths = self.get_texture_paths(mob_type)
+		name = mob_type.lower().replace(" ", "_")
+		if mob_type == "Villager":
+			self.doVillager(mats, texture_paths)
+		elif mob_type == "Zombie":
+			check_entity_texture(context)
+			if self.zombie_variation in ["ZOMBIE", "HUSK"]:
+				# Using Player model so convert player skin to 1.8 format
+				# loadSkinFile()
+				pass
+			else:
+				# Assign textures materials for drown
+				pass
+		elif mob_type == "Skeleton":
+			pass
+		
+		elif mob_type in ("Allay", "Vex"):
+			if mob_type == "Allay":
+				name += "/" + name
+			else:
+				name = "illager"
+			
+		else:
+			pass
+		return {'FINISHED'}
+
+	def doVillager(self, materials: List[Material], texture_paths: List[str]):
```

Comment:
Nitpick about Python convention, use snake_case for functions.
```suggestion
	def do_villager(self, materials: List[Material], texture_paths: List[str]):
```

Reply:
Agreed with above comment, but going further - is there some way that such a function like this could be generalized (and not specify villager in the name)?

Imagine that we somehow overhauled all our rigs, or provided different rigs for 3.0+ (for asset browser) vs 2.8/2.9. I would rather we avoid a situation where we have to change MCprep code just because rigs changed.

So, I think these specialized functions can still exist, I would just rather we try to make them generally useable and if anything rely on property assignments we check for to make them useable (I'm happy to change the rigs to match as needed, remember we want to save them in 2.80 specifically to be forwards compatible)